src/js/actions/superselect.js

```diff
@@ -636,43 +635,44 @@ define(function (require, exports) {
             return adapterOS.postEvent(dragEvent);
         }
         
+        var dontDeselect = modifiers.shift;
         if (dontDeselect) {
             return this.dispatchAsync(events.panel.SUPERSELECT_MARQUEE, { x: x, y: y, enabled: true });
-        } else {
-            var uiStore = this.flux.store("ui"),
-                canvasCoords = uiStore.transformWindowToCanvas(x, y),
-                coveredLayers = _getContainingLayerBounds.call(this, doc.layers, canvasCoords.x, canvasCoords.y);
+        }
 
-            if (coveredLayers.isEmpty()) {
-                // This general bounds check on JS prevents a PS call and starts marquee faster
-                return this.dispatchAsync(events.panel.SUPERSELECT_MARQUEE, { x: x, y: y, enabled: true });
-            } else {
-                // Hide transform controls
-                return descriptor.playObject(toolLib.setToolOptions("moveTool", { "$Abbx": false }))
-                    .bind(this)
-                    .then(function () {
-                        return this.transfer(click, doc, x, y, diveIn, false);
-                    })
-                    .then(function (anySelected) {
-                        if (anySelected) {
-                            var dragEvent = {
-                                eventKind: eventKind,
-                                location: coordinates,
-                                modifiers: dragModifiers
-                            };
-
-                            return adapterOS.postEvent(dragEvent);
-                        }
-                    })
-                    .catch(function () {}) // Move should silently fail if there are no selected layers
-                    .finally(function () {
-                        // Re show the transform controls, which will appear once we're out of the modal state
-                        return descriptor.playObject(
-                            toolLib.setToolOptions("moveTool", { "$Abbx": true }),
-                            { canExecuteWhileModal: true });
-                    });
-            }
+        var uiStore = this.flux.store("ui"),
```

Comment:
There are actually no substantive differences here; I just flattened the control structure a little.